[RFC] Proposal to merge the Patatrack development into CMSSW 10.5.x [squashed]

fwyzard opened the pull request:
This should be equivalent to #25353 after fixing conflicts and **squashing** on top of CMSSW_10_4_0 .

Currently, not all the developments we would like to integrate have been merged in the development branch: see cms-patatrack#200 for the status of the Patatrack branch with respect to readiness for merging upstream.

cmsbuild commented:
The code-checks are being triggered in jenkins.

fwyzard commented:
enable GPU

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/7978

- This PR adds an extra 1104KB to repository

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/7978/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/7978/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly

cmsbuild commented:
The code-checks are being triggered in jenkins.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/7980

- This PR adds an extra 1104KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @fwyzard (Andrea Bocci) for master.

It involves the following packages:

CUDADataFormats/Common
CUDADataFormats/SiPixelCluster
CUDADataFormats/SiPixelDigi
CalibTracker/Records
CalibTracker/SiPixelESProducers
CondFormats/SiPixelObjects
Configuration/Applications
Configuration/ProcessModifiers
Configuration/PyReleaseValidation
Configuration/StandardSequences
DataFormats/GeometrySurface
EventFilter/SiPixelRawToDigi
FastSimulation/Tracking
Geometry/TrackerGeometryBuilder
HeterogeneousCore/CUDACore
HeterogeneousCore/CUDAServices
HeterogeneousCore/CUDAUtilities
HeterogeneousCore/MPICore
HeterogeneousCore/MPIServices
HeterogeneousCore/Producer
HeterogeneousCore/Product
RecoLocalTracker/Configuration
RecoLocalTracker/SiPixelClusterizer
RecoLocalTracker/SiPixelRecHits
RecoPixelVertexing/Configuration
RecoPixelVertexing/PixelTrackFitting
RecoPixelVertexing/PixelTriplets
RecoPixelVertexing/PixelVertexFinding
SimTracker/TrackerHitAssociation
Validation/RecoTrack


The following packages do not have a category, yet:

CUDADataFormats/Common
CUDADataFormats/SiPixelCluster
CUDADataFormats/SiPixelDigi
HeterogeneousCore/CUDACore
HeterogeneousCore/CUDAUtilities
HeterogeneousCore/MPICore
HeterogeneousCore/MPIServices
HeterogeneousCore/Producer
HeterogeneousCore/Product
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please review it and eventually sign? Thanks.
@echabert, @felicepantaleo, @forthommel, @pieterdavid, @Martin-Grunewald, @threus, @mmusich, @seemasharmafnal, @venturia, @makortel, @yduhm, @GiacomoSguazzoni, @rovere, @VinInn, @tocheng, @wmtford, @alesaggio, @ebrondol, @mtosi, @dgulhan, @gbenelli, @dkotlins, @gpetruc, @matt-komm this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smuzaffar commented:
@fwyzard , jenkins job for GPU based testing is failing right now ( https://cmssdt.cern.ch/jenkins/job/ib-any-integration/32576/console ) . I am looking in to it and retry the job (directly via jenkins once fixed). Normal PR testing is running though

cmsbuild commented:
-1

Tested at: 451ec31a049cac1a2c05527d29726295e10f53ba

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32581/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found compilation warning when building:  See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@VinInn please make a PR for Patatrack to fix the build warnings:
```
RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:831:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:835:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:879:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:831:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:835:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:879:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:830:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:834:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:878:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:831:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:835:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
/data/cmsbuild/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-14-1100/src/RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h:879:0: warning: ignoring #pragma unroll  [-Wunknown-pragmas]
```

cmsbuild commented:
-1

Tested at: 451ec31a049cac1a2c05527d29726295e10f53ba

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32574/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation warning when building:  See details on the summary page.

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8047

- This PR adds an extra 1112KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25647 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please check and sign again.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8049

- This PR adds an extra 1116KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25647 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please check and sign again.

cmsbuild commented:
-1

Tested at: b03a690d2de209d184f806c92382b7fd2e99ef4d

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32660/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test phase1PixelTopology_t had ERRORS
---> test HistoContainer_t had ERRORS
---> test cudaAssert_debug had ERRORS
---> test cudaAssert_t had ERRORS


* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: 36a617d7a82100e2d8723b9f18bb2f77b69ed791

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32661/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 48 COMPILER='llvm compile'

```


<pre>/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_5_X_2019-01-16-2300/src/DataFormats/TrackReco/interface/TrackExtraBase.h(29): warning: integer conversion resulted in a change of sign

/cvmfs/cms-ib.cern.ch/nweek-02559/slc7_amd64_gcc700/external/clhep/2.4.0.0-gnimlf/include/CLHEP/Matrix/SymMatrix.h(239): warning: virtual inline function "CLHEP::HepSymMatrix::num_size" was never defined

In file included from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-16-2300/src/RecoPixelVertexing/PixelVertexFinding/src/PixelVertexHeterogeneousProducer.cc:26:
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-16-2300/src/RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinder.h:18:11: error: unknown type name 'nullptr_t'; did you mean 'std::nullptr_t'?
    OnGPU(nullptr_t):
          ^~~~~~~~~
          std::nullptr_t
/cvmfs/cms-ib.cern.ch/nweek-02559/slc7_amd64_gcc700/external/gcc/7.0.0-omkpbe2/lib/gcc/x86_64-unknown-linux-gnu/7.3.1/../../../../include/c++/7.3.1/x86_64-unknown-linux-gnu/bits/c++config.h:235:29: note: 'std::nullptr_t' declared here
  typedef decltype(nullptr)     nullptr_t;
</pre>

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 36a617d7a82100e2d8723b9f18bb2f77b69ed791

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32674/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test phase1PixelTopology_t had ERRORS
---> test HistoContainer_t had ERRORS
---> test cudaAssert_t had ERRORS
---> test cudaAssert_debug had ERRORS


* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: 36a617d7a82100e2d8723b9f18bb2f77b69ed791

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32672/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```


<pre>        ^
/cvmfs/cms-ib.cern.ch/nweek-02559/slc7_amd64_gcc700/external/cuda-api-wrappers/20180504-dlmfga/include/cuda/api/device.hpp:363:53: note: in instantiation of template class 'cuda::device_t<false>' requested here
                return device::peer_to_peer::can_access(id(), peer.id());
                                                                  ^
In file included from /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-17-1100/src/RecoPixelVertexing/PixelVertexFinding/src/PixelVertexHeterogeneousProducer.cc:26:
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_5_X_2019-01-17-1100/src/RecoPixelVertexing/PixelVertexFinding/src/gpuVertexFinder.h:18:11: error: unknown type name 'nullptr_t'; did you mean 'std::nullptr_t'?
    OnGPU(nullptr_t):
          ^~~~~~~~~
          std::nullptr_t
/cvmfs/cms-ib.cern.ch/nweek-02559/slc7_amd64_gcc700/external/gcc/7.0.0-omkpbe2/lib/gcc/x86_64-unknown-linux-gnu/7.3.1/../../../../include/c++/7.3.1/x86_64-unknown-linux-gnu/bits/c++config.h:235:29: note: 'std::nullptr_t' declared here
  typedef decltype(nullptr)     nullptr_t;
</pre>

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8073

- This PR adds an extra 1128KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25647 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please check and sign again.

cmsbuild commented:
-1

Tested at: b3201fd460c5024b3d3ca97aee43815d55cf19d8

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32689/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test phase1PixelTopology_t had ERRORS
---> test HistoContainer_t had ERRORS
---> test cudaAssert_debug had ERRORS
---> test cudaAssert_t had ERRORS


* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@smuzaffar , do you know why the **cms-bot/GPU** claims that there are  Errors Found in the Matrix Tests Outputs ?
The [logs](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32689/runTheMatrix-results/runall-report-step123-.log) seem fine:
> ```
> 10824.8_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_pixelTrackingOnlyGPU_2018+HARVESTFull_pixelTrackingOnlyGPU_2018 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Fri Jan 18 11:30:39 2019-date Fri Jan 18 11:22:02 2019; exit: 0 0 0 0
> 1 1 1 1 tests passed, 0 0 0 0 failed
> ```

cmsbuild commented:
-1

Tested at: b3201fd460c5024b3d3ca97aee43815d55cf19d8

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32691/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

cmsbuild commented:
The code-checks are being triggered in jenkins.

fwyzard commented:
@cmsbuild, please test

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8085

- This PR adds an extra 1128KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25647 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please check and sign again.

cmsbuild commented:
-1

Tested at: b0a465ffeb998c6aa1b15f13f1bac6095994fd08

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32702/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 64 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: b0a465ffeb998c6aa1b15f13f1bac6095994fd08

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32703/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8092

- This PR adds an extra 1152KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 5165cfc9050c4b407c0e6f003d8d60119bc65852

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32715/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: 5165cfc9050c4b407c0e6f003d8d60119bc65852

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32714/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8106

- This PR adds an extra 1156KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25647 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @pohsun, @perrotta, @civanch, @zhenhu, @cmsbuild, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @ssekmen, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @fabiocos, @prebello, @kmaeshima, @pgunnell, @franzoni, @lpernie can you please check and sign again.

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: fdb21bdf8bccb73e1c4b6b63203a42b4bc188bcc

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32734/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

fwyzard commented:
@smuzaffar I think I have fixed every possible problem reported by the bot.
Can you please check why this
```
10824.8_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_pixelTrackingOnlyGPU_2018+HARVESTFull_pixelTrackingOnlyGPU_2018 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Jan 21 15:17:10 2019-date Mon Jan 21 15:08:36 2019; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed
```
is considered as `Errors Found` ?

smuzaffar commented:
@fwyzard , I see these[a]  error messages in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32734/runTheMatrix-results/matrixTests.log . As we search for word `ERROR ` in the log file that is why bot marks it as failure. By the way, the command run by bot is `runTheMatrix.py --job-reports -j 38 -l 10824.8` , you canm run it locally to check why it is causing these matrix errors.

```
ignoring relval_upgrade from default matrix
processing relval_2017
ERROR in Matrix
overwritting 10824.7 not allowed
ERROR in Matrix
overwritting 11224.7 not allowed
ERROR in Matrix
overwritting 10824.7 not allowed
```

fwyzard commented:
Thanks, I'm looking at it.

fwyzard commented:
@smuzaffar I run the same runTheMatrix command, `runTheMatrix.py --job-reports -j 38 -l 10824.8` locally, but I didn't get those ERROR messages.

Any suggestions ?

fwyzard commented:
@makortel ...

cmsbuild commented:
-1

Tested at: fdb21bdf8bccb73e1c4b6b63203a42b4bc188bcc

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25647/32735/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

Dr15Jones commented on HeterogeneousCore/CUDACore/src/GPUCuda.cc:
```diff
@@ -0,0 +1,105 @@
+                                        try {
+                                          auto error = cudaGetErrorName(status);
+                                          auto message = cudaGetErrorString(status);
+                                          throw cms::Exception("CUDAError") << "Callback of CUDA stream " << streamId << " in device " << deviceId << " error " << error << ": " << message;
```
I think this can be replaced with `std::make_exception_ptr`.

fwyzard replied:
we were using that before,  but I changed to a `throw` in order to be able to `catch throw` it in GDB

Dr15Jones replied:
You could change `catch(..)` to `catch(cms::Exception const&)` instead, since you know the type.

fwyzard replied:
ah, I see what you mean - sure, that makes sense.

Dr15Jones commented:
> @Dr15Jones @makortel what should we do regarding the coding rules violations reported here, regarding the use of `catch(...)` in `HeterogeneousCore/Producer/src/HeterogeneousEDProducer.cc` and `HeterogeneousCore/CUDACore/src/GPUCuda.cc` ?

I think we could suppress it using https://clang.llvm.org/extra/clang-tidy/#id3
@gartung will the clang-tidy suppression also work with our static analyzer changes?

fwyzard commented:
For the moment I have added a whitelist to out python checks, same as for the `FWCore` packages.

fwyzard commented:
by the way @kpedro88, do you have any idea why I do not get the same error when testing locally, on top of the latest IB ?

gartung commented:
> @gartung will the clang-tidy suppression also work with our static analyzer changes?
@Dr15Jones not that I know of.

fwyzard committed:
Improve the handling of modules and events

Take into account the proper number of modules: increment by one
pathsAndConsumes.allModules().size(), because it does not include the
"source" module.

Add asserts to check that all ranges are properly closed.

Optionally, delay starting the profiler until after the first event on
each stream has completed. This requires running 'nvprof' with the
'--profile-from-start off' option.

fwyzard committed:
Squash Patatrack developments on top of CMSSW_10_5_X

Squash git@github.com:cms-patatrack/cmssw.git/CMSSW_10_4_X_Patatrack on
top of CMSSW_10_5_X_2019-01-23-2300 .

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8173

- This PR adds an extra 1160KB to repository

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8173/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25647/8173/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly

fwyzard commented:
Will try again...
